src/lib.rs

```diff
@@ -50,10 +81,12 @@ fn pid1_handling(child: Option<Child>) -> ! {
                 };
                 (|| {
                     let child = child?;
-                    let pid = pid?;
+                    let child_exit_status = pid?;
```

Comment:
I think this is wrong, or at the very least the naming is confusing.

Reply:
Sorry for the confusing name, I have renamed it. Can you see if that helps ?